t/023-rewrite/duplicate.t

```diff
@@ -0,0 +1,303 @@
+--- config
+    location /t {
+        rewrite_by_lua_block { ngx.print("Hello, Lua!\n") }
+        rewrite_by_lua_block { ngx.print("Hello, again Lua!\n") }
+        content_by_lua_block { return }
+    }
+--- request
+GET /t
+--- response_body
+Hello, Lua!
+
+
+
+=== TEST 3: mix three different styles
+--- config
+    location /t {
+        rewrite_by_lua ' ngx.log(ngx.ERR, "rewrite_by_lua") ';
+        rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite_by_lua_block") }
+        rewrite_by_lua_file html/a.lua;
+    }
+--- user_files
+>>> a.lua
+ngx.log(ngx.ERR, "rewrite_by_lua_file")
+ngx.print("Hello, Lua!\n")
+--- request
+GET /t
+--- response_body
+Hello, Lua!
+--- error_log
+rewrite_by_lua
+rewrite_by_lua_block
+rewrite_by_lua_file
+
+
+
+=== TEST 4: rewrite directives max limit is 10
+--- config
+    location /t {
+        rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at location") }
+        rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at location") }
+        rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at location") }
+        rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at location") }
+        rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at location") }
+        rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at location") }
+        rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at location") }
+        rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at location") }
+        rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at location") }
+        rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at location") }
+        rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at location") }
+        content_by_lua_block { return }
+    }
+--- request
+GET /t
+--- must_die
+--- error_log
+the number of rewrite_by_lua* directives exceeds 10
+
+
+
+=== TEST 5: the first return status > 200
+--- config
+    location /t {
+        rewrite_by_lua_block { ngx.exit(503) }
+        rewrite_by_lua_block { ngx.print("Hello, again Lua!\n") }
+        content_by_lua_block { return }
+    }
+--- request
+GET /t
+--- error_code: 503
+
+
+
+=== TEST 6: the first return status == 0
+--- config
+    location /t {
+        rewrite_by_lua_block {
+          ngx.log(ngx.ERR, "rewrite_by_lua_block first")
+          ngx.exit(0)
+        }
+        rewrite_by_lua_block { ngx.print("Hello, Lua!\n") }
+        content_by_lua_block { return }
+    }
+--- request
+GET /t
+--- response_body
+Hello, Lua!
+--- error_log
+rewrite_by_lua_block first
+
+
+
+=== TEST 7:
+--- config
+    location /t {
+        rewrite_by_lua_block {
+            ngx.log(ngx.ERR, "first rewrite before sleep")
+            ngx.sleep(0.001)
+            ngx.log(ngx.ERR, "first rewrite after sleep")
+            ngx.sleep(0.001)
+        }
+        rewrite_by_lua_block { ngx.log(ngx.ERR, "second rewrite") }
+        rewrite_by_lua_block { ngx.print("Hello, Lua!\n") }
+    }
+--- request
+GET /t
+--- response_body
+Hello, Lua!
+--- error_log
```

Comment:
I think it's better to use `grep_error_log` and `grep_error_log_out` here so that we can ensure there are no duplicate log entries for the same handler. It is a stronger check than `error_log`.